modules/react/src/components/export-video/export-video-panel-preview.js

```diff
@@ -199,37 +185,64 @@ export class ExportVideoPanelPreview extends Component {
-      position: 'relative',
-      overflow: 'auto'
+      position: 'relative'
+    };
+
+    const loaderStyle = {
+      display: this.props.rendering === false ? 'none' : 'flex',
+      position: 'absolute',
+      background: 'rgba(0, 0, 0, 0.5)',
+      width: `${this.props.exportVideoWidth}px`,
+      height: `${this._getContainerHeight()}px`,
+      alignItems: 'center',
+      justifyContent: 'center'
     };
 
     return (
-      <div id="deck-canvas" style={containerStyle}>
-        <DeckGL
-          ref={this.deckRef}
-          viewState={this.props.viewState}
-          id="hubblegl-overlay"
-          layers={deckGlLayers}
-          style={deckStyle}
-          controller={true}
-          glOptions={{stencil: true}}
-          onWebGLInitialized={gl => this.setState({glContext: gl})}
-          onViewStateChange={this.props.setViewState}
-          // onClick={visStateActions.onLayerClick}
-          {...this.props.adapter.getProps(this.deckRef, () => {})}
-        >
-          {this.state.glContext && (
-            <StaticMap
-              ref={this.mapRef}
-              // reuseMaps // Part of default example but causes modal to lose Mapbox tile layer?
-              mapStyle={this.state.mapStyle}
-              preventStyleDiffing={true}
-              gl={this.state.glContext}
-              onLoad={this._onMapLoad}
-            />
-          )}
-        </DeckGL>
-      </div>
+      <WithKeplerUI>
+        {({LoadingSpinner}) => (
+          <>
+            <div id="deck-canvas" style={containerStyle}>
+              <DeckGL
+                ref={this.deckRef}
+                viewState={this.props.viewState}
+                id="hubblegl-overlay"
+                layers={this.createLayers()}
+                style={deckStyle}
+                controller={true}
+                glOptions={{stencil: true}}
+                onWebGLInitialized={gl => this.setState({glContext: gl})}
+                onViewStateChange={this.props.setViewState}
+                // onClick={visStateActions.onLayerClick}
+                {...this.props.adapter.getProps(this.deckRef, () => {})}
+              >
+                {this.state.glContext && (
+                  <StaticMap
+                    ref={this.mapRef}
+                    mapStyle={this.state.mapStyle}
+                    preventStyleDiffing={true}
+                    gl={this.state.glContext}
+                    onLoad={this._onMapLoad}
+                  />
+                )}
+              </DeckGL>
+            </div>
+            <div className="loader" style={loaderStyle}>
+              <LoadingSpinner />
+              {/* TODO change text styling to match Kepler's */}
+              <div
+                className="rendering-percent"
+                style={{color: 'white', position: 'absolute', top: '175px'}}
+              >
+                {Math.floor(
+                  (this.props.adapter.videoCapture.timeMs / this.props.durationMs) * 100
```

Comment:
Some code clean up to do here is to deconstruct the props for readability up at the top of render():
```
const { adapter, durationMs } = this.props;
```
Also, we should make this render-spinner into its own presentation component.

Reply:
Will do in a new PR